misc/openapi-ballerina/modules/openapi-to-ballerina-generator/src/main/java/org/ballerinalang/openapi/model/BallerinaOpenApi.java

```diff
@@ -125,8 +125,51 @@ private void setPaths(OpenAPI openAPI) throws BallerinaOpenApiException {
                     }
                 });
             }
-            paths.add(new AbstractMap.SimpleEntry<>(path.getKey(), balPath));
+            
+            String resourcePath;
+            // Use ballerina string templating if there are path params.
+            if (hasPathParams(path.getValue())) {
+                resourcePath = "string `" + path.getKey().replace("{", "${") + "`";
+            } else {
+                resourcePath = "\"" + path.getKey() + "\"";
+            }
+            
+            paths.add(new AbstractMap.SimpleEntry<>(resourcePath, balPath));
+        }
+    }
+    
+    /**
+     * Checks if there are any path parameters in the operations.
+     *
+     * @param path The path item in the OpenAPI spec.
+     * @return True if there are path parameters, else false.
+     */
+    private boolean hasPathParams(PathItem path) {
+        if (null != path.getParameters() && path.getParameters().size() > 0) {
+            return path.getParameters().stream().anyMatch(parameter -> {
+                if (null == parameter.getIn()) {
+                    return false;
+                }
+                return parameter.getIn().equals("path");
+            });
+        }
+    
+        if (path.readOperations().size() > 0) {
+            return path.readOperations().stream().anyMatch(operation -> {
+                if (null != operation.getParameters() && operation.getParameters().size() > 0) {
+                    return operation.getParameters().stream().anyMatch(parameter -> {
+                        if (null == parameter.getIn()) {
+                            return false;
+                        }
```

Comment:
just a suggestion ... we can sum this up to a one-line I think. Someway like below... 

`parameter.getIn() == null ? false : parameter.getIn().equals("path")`